src/NetTopologySuite.IO.ShapeFile/Shapefile.cs

```diff
@@ -20,15 +20,23 @@ public partial class Shapefile
         /// <returns>The equivalent for the geometry object.</returns>
         public static ShapeGeometryType GetShapeType(Geometry geom)
         {
-            geom = GetNonEmptyGeometry(geom);
+            var fixedGeom = GetNonEmptyGeometry(geom);
 
-            if (geom == null)
+            if (fixedGeom == null)
                 return ShapeGeometryType.NullShape;
 
-            switch (geom.OgcGeometryType)
+            var geomType = fixedGeom.OgcGeometryType;
+            if (geomType == OgcGeometryType.Point &&
+                geom.OgcGeometryType == OgcGeometryType.MultiPoint)
+            {
+                // NOTE: only multipoints handled in shapefile specifications
+                geomType = OgcGeometryType.MultiPoint;
+            }
```

Comment:
This is a really awkward block.

Maybe it should actually be part of what we currently call `GetNonEmptyGeometry`?  Probably change the contract of that method so that it's expected to return a representative `Geometry` which has its own `ShapeGeometryType` values, therefore it would return the original `MultiPoint` if that's what we give it.

Alternatively, maybe we could just change the comment to clarify why we do this a bit better, something like:

```csharp
// NOTE: shapefile specification distinguishes between Point / MultiPoint, but not between LineString / MultiLineString or Polygon / MultiPolygon
```

Reply:
prably this code, with GetNonemptyGeometry, deserve a larger refactoring

Reply:
after an hour spent on failed refactorings, I think  that the improved comment is the best solution :)